python/ray/tune/logger/aim.py

```diff
@@ -0,0 +1,220 @@
+        log_system_params (:obj:`bool`, optional): Enable/Disable logging of system params such as installed packages,
+            git info, environment variables, etc.
+
+    For more arguments please see the Aim documentation: https://aimstack.readthedocs.io/en/latest/refs/sdk.html
+    """
+
+    VALID_HPARAMS = (str, bool, int, float, list, type(None))
+    VALID_NP_HPARAMS = (np.bool8, np.float32, np.float64, np.int32, np.int64)
+
+    def __init__(
+        self,
+        repo: Optional[str] = None,
+        experiment: Optional[str] = None,
+        metrics: Optional[List[str]] = None,
+        as_multirun: Optional[bool] = False,
+        **aim_run_kwargs
+    ):
+        """
+        See help(AimCallback) for more information about parameters.
+        """
+        assert Run is not None, (
+            "aim must be installed!. You can install aim with"
+            " the command: `pip install aim`."
+        )
+        self._repo_path = repo
+        self._experiment_name = experiment
+        assert bool(metrics) is True or metrics is None
+        self._metrics = metrics
+        self._as_multirun = as_multirun
+        self._run_cls = Run
+        self._aim_run_kwargs = aim_run_kwargs
+        self._trial_run: Dict["Trial", Run] = {}
+
+    def _create_run(self, trial: "Trial") -> Run:
+        """
+        Returns:
+            run (:obj:`aim.sdk.Run`): The created aim run for a specific trial.
+        """
+        experiment_dir = str(Path(trial.logdir).parent)
+        run = self._run_cls(
+            repo=self._repo_path or experiment_dir,
+            experiment=self._experiment_name or trial.experiment_dir_name,
+            **self._aim_run_kwargs
+        )
+        if self._as_multirun:
+            run["trial_id"] = trial.trial_id
+        return run
+
+    def log_trial_start(self, trial: "Trial"):
+        if self._as_multirun:
+            if trial in self._trial_run:
+                self._trial_run[trial].close()
+        elif self._trial_run:
+            return
+
+        trial.init_logdir()
+        self._trial_run[trial] = self._create_run(trial)
+
+        # log hyperparameters
+        if trial and trial.evaluated_params:
+            flat_result = flatten_dict(trial.evaluated_params, delimiter="/")
+            scrubbed_result = {
+                k: value
+                for k, value in flat_result.items()
+                if isinstance(value, tuple(VALID_SUMMARY_TYPES))
+            }
+            self._log_hparams(trial, scrubbed_result)
+
+    def log_trial_result(self, iteration: int, trial: "Trial", result: Dict):
+        # create local copy to avoid problems
+        tmp_result = result.copy()
+
+        step = result.get(TIMESTEPS_TOTAL) or result[TRAINING_ITERATION]
+
+        for k in ["config", "pid", "timestamp", TIME_TOTAL_S, TRAINING_ITERATION]:
+            if k in tmp_result:
+                del tmp_result[k]  # not useful to log these
+
+        context = tmp_result.pop("context", None)
```

Comment:
I got an error when using `as_multirun=True`, on the line where `context["trial"]` is being set.

```suggestion
        context = tmp_result.pop("context", {})
```